core/lib/Cardano/Address/Script.hs

```diff
@@ -490,6 +513,8 @@ prettyErrValidateScript = \case
     WrongKeyHash ->
         "The hash of verification key is expected to have "
         <> show credentialHashSize <> " bytes."
+    NotUniformKeyType ->
+        "All key hashes should have either 'addr_shared_vkh' or 'stake_shared_vhk' prefixes."
```

Comment:
Small note: I'd avoid mentioning prefixes here since we also except their `_vk` and `_xvk` variants. Plus, if we change them again, we'll have to change the error message :no_mouth:. Rather, I would say:

> _"All keys of a script must have the same role: either payment or delegation."_